src/java.compiler/share/classes/javax/lang/model/util/Elements.java

```diff
@@ -51,11 +51,28 @@
 public interface Elements {
 
     /**
-     * Returns a package given its fully qualified name if the package is unique in the environment.
-     * If running with modules, all modules in the modules graph are searched for matching packages.
-     *
-     * @param name  fully qualified package name, or an empty string for an unnamed package
-     * @return the specified package, or {@code null} if it cannot be uniquely found
+     * Returns a package given its fully qualified name if the package is uniquely
+     * determinable in the environment.
+     *
+     * If running with modules, packages of the given name are searched in a
+     * two-stage process:
+     * <ul>
+     *     <li>find non-empty packages with the given name returned by
+     *         {@link #getPackageElement(ModuleElement, CharSequence)},
+     *         where the provided ModuleSymbol is any root module,
```

Comment:
Does "root module" mean one of the modules in the result of Elements.getAllModuleElements? If so, I suggest making "root module" a link to the method.

Reply:
Root module is meant in the java.lang.module sense, i.e. the set of modules that are being compiled. In a new revision, I added a link to the appropriate package javadoc in java.lang.module; and a link to getAllModules for "all modules" in the next point. Does this make sense?

I've generated the javadoc for convenience here:
http://cr.openjdk.java.net/~jlahoda/8236842/docsv2/api/java.compiler/javax/lang/model/util/Elements.html#getTypeElement(java.lang.CharSequence)
http://cr.openjdk.java.net/~jlahoda/8236842/docsv2/api/java.compiler/javax/lang/model/util/Elements.html#getPackageElement(java.lang.CharSequence)